public/app/plugins/datasource/elasticsearch/components/QueryEditor/QueryEditorRow.tsx

```diff
@@ -24,7 +23,7 @@ export const QueryEditorRow: FunctionComponent<Props> = ({
 
   return (
     <fieldset className={styles.root}>
-      <div className={getInlineLabelStyles(theme, 17).label}>
+      <div className={styles.wrapper}>
```

Comment:
not sure why we need these custom styles, can't we use InlineFieldRow & InlineFieldLabel here?

Reply:
They don't really fit this use case here.

this will render as `fieldset` for the entire row (which uses other `InlineField`s & `Segment`s internally), described by (using what would be the `InlineLabel`) a `legend` element.

semantically this should make more sense than having an `InlineLabel` (inside which also we can't use the 2 icon buttons for remove & hide because the first one would be labeled with, for example `Metric (2)` which is incorrect). 

The label element should be used to describe a single input element, while here we need to describe a group of fields.

what do you think? maybe it makes sense to have this in `@grafana/ui` as well?

Reply:
Not sure I understand the difference, maybe @Clarity-89 knows more, InlineFieldLabel is exactly like a segment from the old form styles (gf-form-label). 

So this should be replacable with 

```jsx
<InlineFieldRow>
  <InlineFieldLabel>
     {label}
{onHideClick && (
          <IconButton
            name={hidden ? 'eye-slash' : 'eye'}
            onClick={onHideClick}
            surface="header"
            size="sm"
            aria-pressed={hidden}
            aria-label="hide metric"
            className={styles.icon}
          />
        )}
        <IconButton
          name="trash-alt"
          surface="header"
          size="sm"
          className={styles.icon}
          onClick={onRemoveClick || noop}
          disabled={!onRemoveClick}
          aria-label="remove metric"
        />
  </InlineFieldLabel>
   {children}
</InlineFieldRow>
```

Reply:
But am unable to get this to work either, there seem to be some important missing components in the new inline forms styles. For example cannot find a component that is similar to gf-form that can wrap many gf-form-labels, InlineField requires a label and single input (child).

Reply:
Close to getting it to work here, https://github.com/grafana/grafana/pull/29782 

but needs more tweaking and updated the inline form styles that could require a lot of changes everywhere the inline forms are used

Reply:
Yeah, inline form components do not support fieldset use, we have `FieldSet` [component](https://github.com/grafana/grafana/blob/master/packages/grafana-ui/src/components/Forms/FieldSet.tsx#L13) for non-inline components, but not for the inline ones.

Reply:
@torkelo @Elfo404  After playing around with it a bit, I think I found a solution. Using `InlineField` and passing the label component via its `label` prop seems to work well. `InlineField` will take care of aligning the components inside of it and applying proper margins/spacing. 
```jsx
  <InlineFieldRow>
      <InlineField
        label={
          <InlineLabel width={17}>
            <legend className={styles.label}>{label}</legend>
            {onHideClick && (
              <IconButton
                name={hidden ? 'eye-slash' : 'eye'}
                onClick={onHideClick}
                surface="header"
                size="sm"
                aria-pressed={hidden}
                aria-label="hide metric"
                className={styles.icon}
              />
            )}
            <IconButton
              name="trash-alt"
              surface="header"
              size="sm"
              className={styles.icon}
              onClick={onRemoveClick || noop}
              disabled={!onRemoveClick}
              aria-label="remove metric"
            />
          </InlineLabel>
        }
      >
        <>{children}</>
      </InlineField>
    </InlineFieldRow>
```

Note that `children` needs to be wrapped into a fragment or smth else as `InlineField` expects one child.
Result:
![Screenshot 2020-12-12 at 12 11 31](https://user-images.githubusercontent.com/8878045/101981105-2edbe280-3c73-11eb-9453-79e5fda80ad5.png)

Reply:
That does not work as everything under children will be seen as a single input and will not "line break" (wrap to next row) on smaller screens. 

For query editors, I think we need to support a list of InlineLabel that can wrap and have the correct margins, but tricky. Not sure what the correct building blocks for this is. 

Tried changing the InlineField to have an optional children, the only problem with that is that InlineField has right margin (which gf-form did not have). This right margin creates a double rigth margin when the InlineField only has a label and no child (input)

Reply:
think the best way is probably to create a new component that is similar to InlineField named InlineSegmentGroup or InlineLabelGroup that only has a bottom margin (so similar to gf-form), that represents a pair or a group of labels that will wrap together.

Reply:
It seems like it doesn't wrap the inputs on smaller screens for the old inline styles either, so that seems like a new behavior that needs to be added to the inline components, but yeah seems tricky.

Reply:
Updated https://github.com/grafana/grafana/pull/29782 with another approach

Reply:
It does wrap gf-form elements inside an inline wrapper